server/src/main/java/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -1171,6 +1178,27 @@ public ResourceCountCheckTask() {
 
         @Override
         protected void runInContext() {
+            GlobalLock lock = GlobalLock.getInternLock("ResourceCheckTask");
```

Comment:
:D I think this warrants renaming the PR :p

Reply:
Done